### ISSUE TITLE
Mouse switches client when a fullscreen game is running

### ISSUE THREAD

**Issue by Crylia** [debauchee/barrier#1783]
### What happened?

While trying to play some games I noticed that its not possible because the mouse will leave the screen when I'm turning. This is really annoying and I wonder if there is a way have the application stay focused. The game runs in windows fullscreen btw.

### Version

v2.4.0

### If applicable, where did you install Barrier from?

apt

### What OSes are you seeing the problem on? (Check all that apply)

Linux

### What OS versions are you using?

Pop!_OS 22.04

**Comment by ForbiddenEra** [comment]
Some games will hide the cursor and sample mouse input in a relative manner while resetting the cursor position to the center of the screen quite often. You can tell if your particular game does this if you try to turn slowly (but still moving the mouse far enough to switch screens) and the issue doesn't happen but does happen if you move quicker (as your mouse cursor would have to move far enough from the center of the screen between whenever the game resets the cursor to the center of the screen)

Just a random comment/pointer; I'm not sure about fixing it or otherwise, in fact I was just checking out Barrier for the first time as an potential alternative to commercial software. 

However, if this is the case for you, implementing a fix may be a bit difficult unless somehow Barrier detects games that do this and blacklists them or somehow disables switching if a game is running or especially full screen. I'm not sure how popular this method of sampling mouse input is, whether it's related to particular engines or input libraries or perhaps older games or otherwise but it is something I've seen before and thought it might be worth mentioning.

**Comment by pramirez** [comment]
@Crylia hi

You could try with a key to lock the keyboard/mouse on the screen you are on, that's how I do it and when the game starts I lock only on the screen I need and that way the mouse doesn't escape to the other computer

**Comment by AngelloDeus** [comment]
I think thats a useful solution, but it would be better if it detected it automatically, I have used two monitors with the same computer and they don't switch unless it isn't on fullscreen, you have to pause the game to be able to move the cursor to the other monitor, I think they should add this feature as it would enhance the experience, the shortcut works nice but if it were automatically locked it would be nice.

**Comment by jarrard** [comment]
YES a option to TICK [Ignore Fullscreen App]  AND perhaps even a option to ignore screen switching with EXE/APP option?

NOTE: I understand this project is now eol and there is a new one called input-leap.

even Synergy doesn't have Wayland figured out. Pretty poor considered the premium they ask.  (using X11 atm of cause)
